storage/interface.go

```diff
@@ -27,7 +27,11 @@ const (
 type ProviderRegistry interface {
 	// StorageProviderTypes returns the storage provider types
 	// contained within this registry.
-	StorageProviderTypes() []ProviderType
+	//
+	// Determining the supported storage providers may be dynamic,
+	// but multiple calls for the same registry should return
```

Comment:
This is really two sentences. 

// Determining the supported storage providers may be dynamic.
// Multiple calls for the same registry should return consistent results.

must rather than should?